File: 2020/golang/day-01.go
package main
import (
    "bufio"
    "fmt"
    "os"
    "strconv"
)

// Part 1
func part1(input []int) (result int) {
    for index1 := 0; index1 < len(input); index1++ {
        for index2 := index1 + 1; index2 < len(input); index2++ {
            if (input[index1] + input[index2] == 2020) {
                result = (input[index1]) * (input[index2])
                break
            }
        }
    }
    return
}

// Part 2
func part2(input []int) (result int) {
    for index1 := 0; index1 < len(input); index1++ {
        for index2 := index1 + 1; index2 < len(input); index2++ {
            for index3 := index2 + 1; index3 < len(input); index3++ {
                if (input[index1] + input[index2] + input[index3] == 2020) {
                    result = (input[index1]) * (input[index2]) * (input[index3])
                    break
                }
            }
        }
    }
    return
}

func main() {
    // Read data from input
    file, _ := os.Open("input.txt")
    defer file.Close()

    scanner := bufio.NewScanner(file)
    var input []int

    for scanner.Scan() {
        num, _ := strconv.Atoi(scanner.Text())
        input = append(input, num)
    }

    // Print answer
    fmt.Println(part1(input))
    fmt.Println(part2(input))
    // [Finished in 368.0393ms]
}

File: 2020/python/day-01.py
# Read from input
with open("input.txt", "r") as input_file:
    _input = input_file.read().splitlines()
    _input = [int(i) for i in _input]

# Part 1
def part_1(_input):
    for i in range(len(_input)):
        for j in range(i + 1, len(_input)):
            if ((_input[i] + _input[j]) == 2020):
                return (_input[i] * _input[j])

# Part 2
def part_2(_input):
    for i in range(len(_input)):
        for j in range(i + 1, len(_input)):
            for k in range(j + 1, len(_input)):
                if ((_input[i] + _input[j] + _input[k]) == 2020):
                    return (_input[i] * _input[j] * _input[k])

# Print answer
print(part_1(_input))
print(part_2(_input))
# [Finished in 461ms]